src/library/module.cpp

```diff
@@ -49,17 +49,29 @@ namespace lean {
 struct olean_header {
     // 5 bytes: magic number
     char marker[5] = {'o', 'l', 'e', 'a', 'n'};
-    // 1 byte: version, currently always `1`
-    uint8_t version = 1;
-    // 42 bytes: build githash, padded with `\0` to the right
-    char githash[42];
+    // 1 byte: version, incremented on structural changes to header
+    uint8_t version = 2;
+    // 1 byte of flags:
+    // - whether persisted bignums use GMP or Lean-native encoding
+    uint8_t flags =
+#ifdef LEAN_USE_GMP
+        0b1;
+#else
+        0b0;
+#endif
+    // 33 bytes: Lean version string, padded with '\0' to the right
+    // e.g. "4.12.0-nightly-2024-10-18"
+    char lean_version[33];
+    // 81b008650766442a0dfa9faa796e4588c9d7d3a1
+    // 40 bytes: build githash, padded with `\0` to the right
```

Comment:
There is no padding here, right?

Reply:
There can be, if the githash is empty. I don't think there is anything syntactically restricting it to be exactly 40 bytes, although it will be for actual lean releases.